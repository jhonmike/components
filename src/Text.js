import React from 'react'
import classnames from 'classnames'
import PropTypes from 'prop-types'

import styles from './text.css'

const TextC = props => {
  const { className, tagName, children, ...otherProps } = props
  const tagNames = ['h1', 'h2', 'h3', 'h4', 'h5', 'h6']
  const tag = tagNames.filter(key => key === tagName)[0]
  const classes = classnames(tag ? styles[tag] : null, className)

  return React.createElement(
    tagName || 'p',
    { ...otherProps, className: classes },
    children
  )
}

TextC.propTypes = {
  className: PropTypes.string,
  tagName: PropTypes.string,
  children: PropTypes.node,
}

export default TextC
