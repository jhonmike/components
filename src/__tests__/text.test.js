import renderer from 'react-test-renderer'
import Text from '../Text'

describe('Text Component', () => {
  it('renders component', () => {
    const tree = renderer.create(<Text>Text</Text>).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders Headings component', () => {
    const tree = renderer.create(<Text tagName="h1">Text</Text>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
