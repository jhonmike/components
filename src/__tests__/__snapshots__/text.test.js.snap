// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component renders Headings component 1`] = `
<h1
  className="h1"
>
  Text
</h1>
`;

exports[`Text Component renders component 1`] = `
<p
  className=""
>
  Text
</p>
`;
